source/plugins/go/src/go.mod

```diff
@@ -3,33 +3,17 @@ module Docker-Provider/source/plugins/go/src
 go 1.14
 
 require (
-	code.cloudfoundry.org/clock v1.0.1-0.20200131002207-86534f4ca3a5 // indirect
 	github.com/Azure/azure-kusto-go v0.3.2
 	github.com/Azure/go-autorest/autorest/azure/auth v0.4.2
+	github.com/Azure/go-autorest/autorest/to v0.4.0 // indirect
+	github.com/dnaeon/go-vcr v1.2.0 // indirect
 	github.com/fluent/fluent-bit-go v0.0.0-20171103221316-c4a158a6e3a7
-	github.com/ghodss/yaml v0.0.0-20150909031657-73d445a93680 // indirect
-	github.com/gogo/protobuf v0.0.0-20170330071051-c0656edd0d9e // indirect
-	github.com/golang/glog v0.0.0-20141105023935-44145f04b68c // indirect
-	github.com/google/btree v0.0.0-20160524151835-7d79101e329e // indirect
-	github.com/google/gofuzz v0.0.0-20161122191042-44d81051d367 // indirect
-	github.com/google/uuid v1.1.1
-	github.com/googleapis/gnostic v0.0.0-20170729233727-0c5108395e2d // indirect
-	github.com/gregjones/httpcache v0.0.0-20170728041850-787624de3eb7 // indirect
-	github.com/json-iterator/go v0.0.0-20180612202835-f2b4162afba3 // indirect
+	github.com/google/uuid v1.1.2
 	github.com/microsoft/ApplicationInsights-Go v0.4.3
-	github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd // indirect
-	github.com/modern-go/reflect2 v0.0.0-20180320133207-05fbef0ca5da // indirect
-	github.com/peterbourgon/diskv v2.0.1+incompatible // indirect
-	github.com/philhofer/fwd v1.0.0 // indirect
-	github.com/satori/go.uuid v1.2.1-0.20181028125025-b2ce2384e17b // indirect
+	github.com/philhofer/fwd v1.1.1 // indirect
 	github.com/tinylib/msgp v1.1.2
 	github.com/ugorji/go v1.1.2-0.20180813092308-00b869d2f4a5
-	golang.org/x/net v0.0.0-20200421231249-e086a090c8fd // indirect
-	golang.org/x/time v0.0.0-20161028155119-f51c12702a4d // indirect
-	gopkg.in/inf.v0 v0.9.0 // indirect
 	gopkg.in/natefinch/lumberjack.v2 v2.0.0-20170531160350-a96e63847dc3
-	k8s.io/api v0.0.0-20180628040859-072894a440bd // indirect
-	k8s.io/apimachinery v0.0.0-20180621070125-103fd098999d
-	k8s.io/client-go v8.0.0+incompatible
-	golang.org/x/crypto v0.0.0-20201216223049-8b5274cf687f
+	k8s.io/apimachinery v0.21.0
+	k8s.io/client-go v0.21.0
```

Comment:
@gracewehner, because of this update, did you find any compatibility issues with k8s versions < 1.17?  is this upgrade agnostic to k8s version?

Reply:
@ganga1980 do we need compatibility for < 1.17? I thought we only support the same Kubernetes versions as AKS, even for non-AKS: https://docs.microsoft.com/en-us/azure/azure-monitor/containers/container-insights-onboard#supported-configurations

Reply:
@gracewehner , yes, our support matrix same as as what you have pointed out. I am trying to understand the impact because of this and based on that, we can decide whether we would need to cover in docs or not as part of this release. From our telemetry in last 24hr, i see we have 5K+ clusters with 1.14, 1.15 & 1.16 together. do you know what will be broken with versions < 1.17 version ?

Reply:
i dont understand where the "1.17" is coming from ? Where do you see this is not compatible with < 1.17 specifically? why not 1.18 ? The fix here is we were using clint go version that was deprecated with k8s 1.11 in 2019. So she is taking the latest version of client go. The only place this is used is when enrichment is turned ON.

Reply:
I see 57 clusters with version < 1.17 and enrichment enabled. The client-go documentation is a bit vague about compatibility: https://github.com/kubernetes/client-go#compatibility-client-go---kubernetes-clusters

Reply:
yes, that's what I was referring..

https://github.com/kubernetes/client-go#client-go
https://github.com/kubernetes/client-go/blob/master/INSTALL.md#using-a-specific-version
because of this client-go ( to tag v0.x.y) update, no issue to the < 1.17 clusters then we are good.  Since the documentation is very vague so I think, better to validate.

Reply:
Thanks Ganga, I have now validated with 1.16 and 1.17 clusters and checked that there aren't errors in `fluent-bit-out-oms-runtime.log` for these clusters